adiachenko/src/App.js

```diff
@@ -13,14 +14,37 @@ const MainTheme = {
 
 import GlobalStyle from "./globalStyles";
 
-function App() {
+const App = () => {
+  const [isSearch, setSearch] = useState(true);
+  const [film, setFilm] = useState(null);
+  const [sortCategory, setSortCategory] = useState('all');
+  const openSearch = () => setSearch(true);
+  const openFilmDetails = () => setSearch(false);
+
+  const defaultValue = {
+      film,
+      sortCategory,
+      setCategory: (category) => {
+          setSortCategory(category);
+      },
+      setFilm: (film) => {
+          setFilm(film)
+          openFilmDetails();
+      }
+  };
+
   return (
       <div className="app">
         <GlobalStyle />
         <ThemeProvider theme={ MainTheme }>
-          <Header />
-          <MainContent />
-          <Footer />
+            <AppContext.Provider value={defaultValue}>
```

Comment:
I would have suggested to [isolate](https://codepen.io/ygorbunkov/pen/LYbMqJG) `<AppContext.Provider />` within a separate component not to overload the main component.